crates/core/component/dex/src/component/router/route_and_fill.rs

```diff
@@ -23,21 +24,14 @@ use super::fill_route::FillError;
 /// a block's batch swap flows.
 #[async_trait]
 pub trait HandleBatchSwaps: StateWrite + Sized {
-    #[instrument(skip(
-        self,
-        trading_pair,
-        batch_data,
-        block_height,
-        epoch_starting_height,
-        params
-    ))]
+    #[instrument(skip(self, trading_pair, batch_data, block_height, epoch, params))]
     async fn handle_batch_swaps(
         self: &mut Arc<Self>,
         trading_pair: TradingPair,
         batch_data: SwapFlow,
         // TODO: why not read these 2 from the state?
         block_height: u64,
-        epoch_starting_height: u64,
+        epoch: Epoch,
```

Comment:
This seems dangerous to me -- I realize that the code is already like this, but given the importance of getting the values right, I think it would be good to eliminate these arguments and to instead have the correct values read out of the state right when the BSOD is created.

Reply:
Changed to read from the state inside this method.

The block height is read from the block in the end_block handler, so that is left unchanged for now.